internal/services/identitygovernance/access_package_catalog_role_assignment_resource_test.go

```diff
@@ -0,0 +1,75 @@
+package identitygovernance_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/go-azure-sdk/sdk/odata"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-azuread/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azuread/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azuread/internal/clients"
+	"github.com/hashicorp/terraform-provider-azuread/internal/utils"
+)
+
+type AccessPackageCatalogRoleAssignmentResource struct{}
+
+func TestAccAccessPackageCatalogRoleAssignmentResource_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azuread_access_package_catalog_role_assignment", "test")
+	r := AccessPackageCatalogRoleAssignmentResource{}
+
+	data.DataSourceTest(t, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("catalog_id").IsUuid(),
+				check.That(data.ResourceName).Key("principal_object_id").IsUuid(),
+				check.That(data.ResourceName).Key("role_id").Exists(),
+			),
+		},
```

Comment:
We'll also want to add an import test here, e.g.

```suggestion
		},
		data.ImportStep(),
```